edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/ModuleAuthentication.cs

```diff
@@ -16,7 +18,18 @@ public ModuleAuthentication(ITokenProvider tokenProvider, string deviceId, strin
             this.tokenProvider = Preconditions.CheckNotNull(tokenProvider);
         }
 
-        protected override Task<string> SafeCreateNewToken(string iotHub, int suggestedTimeToLive) =>
-            this.tokenProvider.GetTokenAsync(Option.Some(TimeSpan.FromSeconds(suggestedTimeToLive)));
+        protected override Task<string> SafeCreateNewToken(string iotHub, int suggestedTimeToLive)
+        {
+            try
+            {
+                return this.tokenProvider.GetTokenAsync(Option.Some(TimeSpan.FromSeconds(suggestedTimeToLive)));
+            }
+            catch(HttpHsmCommunicationException ex)
```

Comment:
Kind of ruins the `ITokenProvider` abstraction if you're hard-coding the exception thrown by a specific `ISignatureProvider` impl used by a specific `ITokenProvider` impl.

On the other hand, the `ITokenProvider` / `ISignatureProvider` impls shouldn't be throwing `IoTHubCommunicationException`s themselves either, because they shouldn't have to care about the context of their caller.

Is there a way to do the equivalent of https://github.com/Azure/iotedge/commit/c3997581d3e53417a0a2253fba6c76f3da5221cb#diff-9473343eaf04e45a06be5030005d16c7R388-R392 ? Eg inherit a new marker interface on `HttpHsmCommunicationException` and `catch` that interface?

Reply:
Good points; Let me think about this a little bit more...

Reply:
OK, I pushed something a little different; let me know what you think. I don't like that it creates two new layers of exceptions, but I like that I'm still able to `catch` only specific errors (instead of catching all and then filtering/re-throwing).